src/main/java/com/crud/library/domain/Book.java

```diff
@@ -0,0 +1,80 @@
+package com.crud.library.domain;
+
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import javax.persistence.*;
+import javax.validation.constraints.NotNull;
+import java.util.ArrayList;
+import java.util.List;
+
+@NoArgsConstructor
+@Getter
+@Setter
+@Entity(name = "book")
+public class Book {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @NotNull
+    private long id;
+
+    @Column(name = "title")
+    private String title;
+
+    @Column(name = "author")
+    private String author;
+
+    @Column(name = "year_of_publication")
+    private int yearOfPublication;
+
+    @OneToMany(
+            targetEntity = CopyOfTheBook.class,
+            mappedBy = "book",
+            cascade = CascadeType.ALL,
+            fetch = FetchType.LAZY
+    )
+    private List<CopyOfTheBook> copiesBookInLibrary = new ArrayList<>();
+
+    public static class BookBuilder {
+        private long id;
+        private String title;
+        private String author;
+        private int yearOfPublication;
+        private List<CopyOfTheBook> copiesBookInLibrary;
+
```

Comment:
dodaj metodę statyczną zwracającą builder `BookBuilder.newBuilder()` . to często spotykana praktyka, lepiej się to czyta.